scripts/openshiftci-presubmit-all-tests.sh

```diff
@@ -2,69 +2,30 @@
 
 # fail if some commands fails
 set -e
-# show commands
-set -x
-
-ARCH=$(uname -m)
-export CI="openshift"
-if [ "${ARCH}" == "s390x" ]; then
-    make configure-installer-tests-cluster-s390x
-elif [ "${ARCH}" == "ppc64le" ]; then
-    make configure-installer-tests-cluster-ppc64le
-else
-    make configure-installer-tests-cluster
+# Hide command that deals with secrets
+set +x
+if [[ -f $ODO_RABBITMQ_AMQP_URL ]]; then
+    export AMQP_URI=$(cat $ODO_RABBITMQ_AMQP_URL)
 fi
-make bin
-mkdir -p $GOPATH/bin
-make goget-ginkgo
-export PATH="$PATH:$(pwd):$GOPATH/bin"
-export ARTIFACTS_DIR="/tmp/artifacts"
-export CUSTOM_HOMEDIR=$ARTIFACTS_DIR
-
-# Copy kubeconfig to temporary kubeconfig file
-# Read and Write permission to temporary kubeconfig file
-TMP_DIR=$(mktemp -d)
-cp $KUBECONFIG $TMP_DIR/kubeconfig
-chmod 640 $TMP_DIR/kubeconfig
-export KUBECONFIG=$TMP_DIR/kubeconfig
 
-# Login as developer
-odo login -u developer -p password@123
+##### These are varialbes used by ci-firewall as one of the ways to get its parameters
+# If AMQP_URI is not set by the time we reach here, show error message and exit.
+export AMQP_URI=${AMQP_URI:?"Please set AMQP_URI env with amqp uri or provide path of file containing it as ODO_RABBITMQ_AMQP_URL env"}
+export JOB_NAME="odo-pr-build"
+export REPO_URL="https://github.com/openshift/odo"
+# Extract PR NUMBER from prow job spec, which is injected by prow.
+export TARGET="$(jq .refs.pulls[0].number <<< $(echo $JOB_SPEC))"
+##### ci-firewall parameters end
 
-# Check login user name for debugging purpose
-oc whoami
+# The version of CI_FIREWALL TO USE
+export CI_FIREWALL_VERSION="valpha"
 
-if [ "${ARCH}" == "s390x" ]; then
-    # Integration tests
-    make test-integration
-    make test-cmd-login-logout
-    make test-cmd-project
-    # E2e tests
-    make test-e2e-all
-elif  [ "${ARCH}" == "ppc64le" ]; then
-    # Integration tests
-    make test-integration
-    make test-cmd-login-logout
-    make test-cmd-project
-    # E2e tests
-    make test-e2e-all
-else
-    # Integration tests
-    make test-integration || error=true
-    make test-integration-devfile || error=true
-    make test-cmd-login-logout || error=true
-    make test-cmd-project || error=true
-    make test-operator-hub || error=true
-
-    # E2e tests
-    make test-e2e-all || error=true
+echo "Getting ci-firewall, see https://github.com,/mohammedzee1000/ci-firewall"
+# show commands
+set -x
 
-    # Fail the build if there is any error while test execution
-    if [ $error ]; then 
-        exit -1
-    fi
-fi
+curl -kLO https://github.com/mohammedzee1000/ci-firewall/releases/download/valpha/ci-firewall-linux-amd64.tar.gz
+tar -xzf ci-firewall-linux-amd64.tar.gz
 
-cp -r reports $ARTIFACTS_DIR 
+./ci-firewall request --runscript scripts/run_all_tests.sh --timeout 2h15m
```

Comment:
Timeout has been set on the basis of what?

Reply:
Tests can take an hour as far as I can see. So I am using a safe timeout. We will reduce it as we move forward.